sdk/evaluation-context.json

```diff
@@ -0,0 +1,147 @@
+{
+    "$defs": {
+        "EnvironmentEvaluationContext": {
+            "properties": {
+                "api_key": {
+                    "title": "Api Key",
+                    "type": "string"
+                }
+            },
+            "required": [
+                "api_key"
+            ],
+            "title": "EnvironmentEvaluationContext",
+            "type": "object"
+        },
+        "FeatureEvaluationContext": {
+            "properties": {
+                "name": {
+                    "title": "Name",
+                    "type": "string"
+                }
+            },
+            "required": [
+                "name"
+            ],
+            "title": "FeatureEvaluationContext",
+            "type": "object"
+        },
+        "IdentityEvaluationContext": {
+            "properties": {
+                "identifier": {
+                    "title": "Identifier",
+                    "type": "string"
+                },
+                "traits": {
+                    "additionalProperties": {
+                        "anyOf": [
+                            {
+                                "$ref": "#/$defs/TraitEvaluationContext"
+                            },
+                            {
+                                "type": "null"
+                            }
+                        ]
+                    },
+                    "title": "Traits",
+                    "type": "object"
```

Comment:
Should this be a list?

Reply:
The idea is that we expect a map of strings (trait keys) to `TraitEvaluationContext | null` (trait values):

1. The schema ensures the trait data is being sent once per trait key.
2. We don't need a nullable `TraitEvaluationContext.value` field — just use `{"trait_key": null}` to delete a persisted trait.

Reply:
I see, ok. I guess what I don't necessarily like about that is that the keys aren't well defined, but perhaps that's acceptable with a schema such as this one. It will be harder to model with API documentation (certainly with DRF at least).

Reply:
Doesn't DRF have `DictField`?

Reply:
> the keys aren't well defined

You don't have much choice in JSON — keys are strings (which is what we need for traits)

Reply:
> Doesn't DRF have `DictField`?

It does, but then I don't think there's a way (without manual intervention) to define the schema of that dict field, so you just end up with a dict-like black hole! 

> You don't have much choice in JSON — keys are strings (which is what we need for traits)

Well, the other choice is still to use a list, right? But I agree that using an object here at least gives us a better way of validating uniqueness.

Reply:
Yep, fair enough. I'm not necessarily advocating that we should definitely use a list, I'm mostly just raising the concern that the API documentation will be slightly more challenging.

Reply:
> It does, but then I don't think there's a way (without manual intervention) to define the schema of that dict field, so you just end up with a dict-like black hole!

DRF docs state the opposite: https://www.django-rest-framework.org/api-guide/fields/#dictfield

Basically, `child = fields.Nested(TraitEvaluationContextSchema)` will do the trick.

Reply:
Besides the above, we could probably use a Pydantic model or validate directly against the json schema.

Reply:
This is all fine, and think I do agree that it's the best option here, but it's worth noting that it's not as simple as `child = fields.Nested(TraitEvaluationContextSchema)` because of the fact that the keys are dynamic.

Reply:
Sorry for dragging this on, but it is, unless I'm massively overlooking something?

According to docs, `child` is not a key name, it's a meta field representing every dict value.

Reply:
... oh, my bad! Sounds good then :)